docs/docusaurus.config.js

```diff
@@ -5,63 +5,45 @@ const {
   remarkPlugins: themeRemarkPlugins,
 } = require('@rasahq/docusaurus-theme-tabula');
 
+const isDev = process.env.NODE_ENV === 'development';
+const isStaging = process.env.NETLIFY && process.env.CONTEXT === 'staging';
+const isPreview = process.env.NETLIFY && process.env.CONTEXT === 'deploy-preview';
 
 // FIXME: remove "next/" when releasing + remove the "next/" in
-// http://github.com/RasaHQ/rasa-website/blob/master/netlify.toml
+// https://github.com/RasaHQ/rasa-website/blob/master/netlify.toml
 const BASE_URL = '/docs/rasa/next/';
 const SITE_URL = 'https://rasa.com';
-
 // NOTE: this allows switching between local dev instances of rasa/rasa-x
-const isDev = process.env.NODE_ENV === 'development';
 const SWAP_URL = isDev ? 'http://localhost:3001' : SITE_URL;
 
-/* VERSIONING: WIP */
+let existingVersions = [];
+try { existingVersions = require('./versions.json'); } catch (e) { console.info('no versions.json file found') }
+const currentVersionPath = isDev || isPreview ? '/' : `${existingVersions[0]}/`;
 
 const routeBasePath = '/';
-let versions = [];
-try { versions = require('./versions.json'); } catch (ex) { console.info('no versions.json file found; assuming dev mode.') }
+const existingVersionRE = new RegExp(
+  `${routeBasePath}/(${existingVersions.reduce((s, v, i) => `${s}${i > 0 ? '|' : ''}${v}`, '')}).?`,
+);
+const currentVersionRE = new RegExp(`(${routeBasePath})(.?)`);
 
-const legacyVersion = {
-  label: 'Legacy 1.x',
-  href: 'https://legacy-docs-v1.rasa.com',
-  target: '_self',
+const versionLabels = {
+  current:
+    isDev || isPreview
+      ? `Next (${isPreview ? 'deploy preview' : 'dev'})`
+      : existingVersions.length < 1
+      ? 'Current'
+      : 'Next',
```

Comment:
Is this how we're switching it from current to next when there are multiple versions? If so, i'd prefer it it were `Master` (as i believe that's what it is going to be, right?)

Reply:
Yes, this is already fixed. I picked up Maxime's label "Master/Unreleased"